query/src/planners/plan_scheduler.rs

```diff
@@ -44,18 +45,37 @@ impl PlanScheduler {
         let mut results = vec![];
         let cluster = ctx.try_get_cluster()?;
         let cluster_nodes = cluster.get_nodes()?;
-        let cluster_nums = if cluster_nodes.is_empty() {
-            1
+        let priority_sum = if cluster_nodes.is_empty() {
+            0
         } else {
-            cluster_nodes.len()
+            cluster_nodes.iter().map(|n| n.priority as usize).sum()
         };
 
-        // Align to the cluster numbers.
-        let remainder = partitions.len() % cluster_nums;
-        let chunk_size = (partitions.len() + remainder) / cluster_nums + 1;
-        for chunks in partitions.chunks(chunk_size) {
+        let total_chunks = partitions.len();
+        let mut index = 0;
+        let mut num_chunks_so_far = 0;
+        let mut chunk_size;
+
+        while num_chunks_so_far < total_chunks {
```

Comment:
There are some problems here, although the total chunks is evenly divided according to priority, but it is not assigned according to prigionty.
As an example:
If we have 100 partitions and one cluster with 2-nodes:
node1: {..., priority: 7}
node2: {..., priority: 3}

The partitions distribution should be: 
node1: 7/(7+3)
node2: 3/(7+3)

Reply:
hmm, give a deep check , it seems works as the above comments.
Well,  it's better to add an unit test to show different priority has different partitions.

Reply:
I added a unit test called "test_scheduler_plan_with_3_nodes_diff_priority"